### PR TITLE
#109 Alexandr Senyuk

### PR DESCRIPTION
**Pull request by alexandr-101**

### CODE REVIEW

**Comment by alexandr-101** [comment]
#109 Alexandr Senyuk
bot.rb without effensens

File: triangle.rb [hunk 1]
```diff
@@ -14,7 +14,18 @@
 #   about_triangle_project_2.rb
 #
 def triangle(a, b, c)
```
Comment (anatoliliotych) [inline triangle.rb]: https://github.com/bbatsov/ruby-style-guide

Please re-read style guide

File: bot.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
+require 'telegram/bot'
```
Comment (anatoliliotych) [inline bot.rb]: Please make a separate repo

File: about_scoring_project.rb [hunk 3]
```diff
@@ -30,7 +30,24 @@
...
+      count -=3
+    end
+    if (count >= 3 and dig == 1) 
+      points += 1000
+      count -=3
+    end
+    points += 50*count if dig == 5
+    points += 100*count if dig == 1
+  end
+  return points
```
Comment (anatoliliotych) [inline about_scoring_project.rb]: no return if it's the last line

File: about_scoring_project.rb [hunk 4]
```diff
@@ -30,7 +30,24 @@
 # Your goal is to write the score method.
 
 def score(dice)
```
Comment (anatoliliotych) [inline about_scoring_project.rb]: check style guide

File: about_scoring_project.rb [hunk 5]
```diff
@@ -30,7 +30,24 @@
...
-  # You need to write this method
+  points = 0
+  counter = {1=>0, 2=>0, 3=>0, 4=>0, 5=>0, 6=>0}
```
Comment (anatoliliotych) [inline about_scoring_project.rb]: try to find more elegant solution.

File: about_scoring_project.rb [hunk 6]
```diff
@@ -30,7 +30,24 @@
...
+  dice.map do |dig, count|
```
Comment (anatoliliotych) [inline about_scoring_project.rb]: it could be done without such complexity

File: bot.rb (hunk 2 above)
Comment (anatoliliotych) [inline bot.rb]: Move bot to another repository please!

File: about_scoring_project.rb [hunk 7]
```diff
@@ -30,7 +30,14 @@
 # Your goal is to write the score method.
 
 def score(dice)
-  # You need to write this method
+  points = 0
+  (1..6).times do |elem| 
+    counter = dice.count(elem)
+    elem = 10 unless elem == 1
+    points += elem * 100; counter -=3 unless counter >= 3
+    points += elem *10 * (counter
```
Comment (anatoliliotych) [inline about_scoring_project.rb]: where is ```)``` ???

File: about_scoring_project.rb [hunk 8]
```diff
@@ -30,7 +30,14 @@
 # Your goal is to write the score method.
 
 def score(dice)
-  # You need to write this method
+  points = 0
+  (1..6).times do |elem| 
+    counter = dice.count(elem)
+    elem = 10 unless elem == 1
+    points += elem * 100; counter -=3 unless counter >= 3
```
Comment (anatoliliotych) [inline about_scoring_project.rb]: try to not use unless with two lines

File: about_dice_project.rb [hunk 9]
```diff
@@ -2,9 +2,13 @@
...
 # Implement a DiceSet Class here:
 #
-# class DiceSet
-#   code ...
-# end
+class DiceSet
+  attr_reader :values
+
+  def roll (n)
+    @values = Array.new(n){rand(5)+1}
```
Comment (anatoliliotych) [inline about_dice_project.rb]: ``` rand(1..6) ``` looks better

**Pull request closed without merging**